packages/autocomplete-core/src/utils.ts

```diff
@@ -44,14 +45,25 @@ function normalizeSource<TItem>(
   };
 }
 
+const runConcurrentSafePromiseForGetSources = createConcurrentSafePromise<
+  Array<AutocompleteSource<any>>
+>();
+const runConcurrentSafePromiseForSource = createConcurrentSafePromise<
+  InternalAutocompleteSource<any>
+>();
+
 export function getNormalizedSources<TItem>(
   getSources: AutocompleteOptions<TItem>['getSources'],
   options
 ): Promise<Array<InternalAutocompleteSource<TItem>>> {
-  return Promise.resolve(getSources(options)).then((sources) =>
+  return runConcurrentSafePromiseForGetSources(
+    Promise.resolve(getSources(options))
+  ).then((sources) =>
     Promise.all(
       sources.filter(Boolean).map((source) => {
-        return Promise.resolve(normalizeSource<TItem>(source));
+        return runConcurrentSafePromiseForSource(
```

Comment:
If I understand correctly, don't we need to have a difference instance of `createConcurrentSafePromise` per source here? For each source, we want to keep the last result, not the last result of all the sources.
Let me know if I'm misunderstanding!

Reply:
Yeah exactly. This case is complex because of the dynamic nature of sources. This PR doesn't cover this case anymore.